### PR TITLE
Update documentation to leverage IPAM and DNS

### LINKED ISSUE #471: make use of DNS throughout the documentation
ATM the weaveDNS documentation is completely separate from the main documentation, with just the odd reference to it from the latter. This was desirable when weaveDNS was new and experimental, but now it actually hinders adoption.

The weave 1.0 walkthrough in #242 presents a much more integrated view of weaveDNS, and hence can serve as guidance for this issue, though we won't be able to go quite as far until a few more features have landed.

### LINKED ISSUE #869: make use of default IPAM throughout documentation
IPAM should be documented as an integral rather than separate feature. i.e. throughout the docs we should generally make use of IPAM. Usually with the default sub-net, i.e. not specifying any IP/network. And specifying a subnet (but no IP) should become the primary means to deal with multiple subnets. I reckon specifying individual IP addresses may well be something we only cover once, in a separate section.

We will still want to keep docs specifically about IPAM, to explain stuff about quorum and cover various exotic uses.

### PR DESCRIPTION
**Pull request by awh**
@rade @bboreham I'm not expecting this to be actually merged yet, I'm just using it to make changes available to you ASAP to get early feedback; I'll continue pushing self contained updates to other areas of the documentation + incorporate your feedback as you make it.

This PR updates just the main README initially - examples look much much better! Any advance on the rather boring h1c1/h2c1 container names? They're a bit mechanistic... something fun that we can carry through as a theme would be better IMO.

Fixes #869. Fixes #471. (eventually...)

### CODE REVIEW

**Comment by bboreham** [comment]
For the README I think c1 and c2 would work fine.  Is there a point where there is an h1c2 or h2c2?

File: README.md [hunk 1]
```diff
@@ -56,47 +56,41 @@ two containers, one on each host.
...
+    host1$ weave run --name h1c1 -t -i ubuntu
 
-The first line starts the weave router, in a container. This needs to
-be done once on each host. The required docker image for the weave
-router container is downloaded automatically. There is also a `weave
-setup` command for downloading this and other images required for
+The first two lines start the weave router and DNS, each inside their
+own container - this needs to be done once per host. Whilst the
+required docker images are downloaded automatically, there is also a
+`weave setup` command for downloading all the images required for
```
Comment (bboreham) [inline README.md]: `weave setup` could go elsewhere; it is not a key feature to learn about on first contact.

**Comment by awh** [comment]
I've updated the manual addressing feature so it's a replay of the first example - this means that the following application subnet isolation makes sense without changes (unless there is a way that can be IPAMified - thoughts?)

File: site/features.md [hunk 2]
```diff
@@ -55,6 +56,49 @@ now re-use the same tools and techniques when deploying applications
 as containers as we would have done when deploying them 'on metal' in
 our data centre.
 
+### <a name="manual-addressing"></a>Manual addressing
+
+The examples so far have leveraged weave's [address allocation](#ipam)
+and [discovery](#dns) features, meaning that we've not had to concern
+ourselves with IP addresses at all. It is however possible to specify
```
Comment (bboreham) [inline site/features.md]: "It is, however, possible"
but "possible" plus "if you want" seems overkill.

**Comment by bboreham** [comment]
You can request IPAM allocate an address in a subnet, e.g. `net:10.180.0.0/16`

File: site/features.md [hunk 3]
```diff
@@ -174,31 +232,38 @@ anywhere.
...
 
-    host2$ weave expose 10.2.1.102/24
+    host2$ weave expose
 
-choosing an unused IP address in the application subnet. (There is a
-corresponding 'hide' command to revert this step.)
+(There is a corresponding 'hide' command to revert this step.)
 
-Now
+Now,  after finding allocated IPs via `weave ps`
```
Comment (awh) [inline site/features.md]: This is a bit janky, but weaveDNS names won't resolve from the host...

**Comment by bboreham** [comment]
LGTM, apart from the names h1c2 etc., which I still find ugly.

**Comment by awh** [comment]
@rade suggests a1, a2 and b1, b2 (letter == subnet, number == host) - thoughts @bboreham?

**Comment by bboreham** [comment]
Better.  Or you could start with c1, c2 as before and then move to b1, b2 when you have a second subnet?

**Pull request merged**